lib/spack/llnl/util/filesystem.py

```diff
@@ -152,6 +152,31 @@ def lookup(name):
     shutil.copystat = copystat
 
 
+def polite_path(components):
```

Comment:
Can you change the function signature to something more sensible? Either follow os.path.join(...) style or typehint iterable[str], cause right now it also accepts strings